src/main/java/ca/on/oicr/pineryreports/reports/impl/LanesBillingReport.java

```diff
@@ -0,0 +1,464 @@
+    return lanes;
+  }
+  
+  private void addSummaryLanes(Map<String, BigDecimal> lanes, String laneType, BigDecimal laneCount) {
+    lanes.merge(laneType, laneCount, BigDecimal::add);
+  }
+  
+  static final List<Map<String, Map<String, BigDecimal>>> listifySummary(Map<String, Map<String, BigDecimal>> summary) {
+    // need to convert it to a list, because getRow() takes an index and the treemap doesn't yet have one of those
+    List<Map<String, Map<String, BigDecimal>>> regrettable = new ArrayList<Map<String, Map<String, BigDecimal>>>();
+    
+    for (Map.Entry<String, Map<String, BigDecimal>> entry : summary.entrySet()) {
+      Map<String, Map<String, BigDecimal>> oneProjModel = new HashMap<>();
+      oneProjModel.put(entry.getKey(), entry.getValue());
+      regrettable.add(oneProjModel);
+    }
+    return regrettable;
+  }
+
+  @Override
+  protected List<ColumnDefinition> getColumns() {
+    return Arrays.asList(
+        new ColumnDefinition("Project"),
+        new ColumnDefinition("Instrument Model"),
+        new ColumnDefinition(DNA_LANE),
+        new ColumnDefinition(RNA_LANE),
+        new ColumnDefinition(MIXED_LANE),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition("")
+    );
+  }
+  
+  private List<String> getDetailedHeadings() {
+    return Arrays.asList(
+        "Project",
+        "Instrument Name",
+        "Instrument Model",
+        "Run Status",
+        "Run End Date",
+        "Run Name",
+        "Lane",
+        DNA_LANE,
+        RNA_LANE,
+        MIXED_LANE
+    );
+  }
+
+  @Override
+  protected int getRowCount() {
+    return completed.getDetailedData().size()
+        + (completed.getDetailedData().size() > 0 ? 2 : 0)
+        + completed.getSummaryAsList().size()
+        + (failed.getDetailedData().size() > 0 ? 2 : 0)
+        + failed.getDetailedData().size()
+        + (failed.getDetailedData().size() > 0 ? 2 : 0)
+        + failed.getSummaryAsList().size();
+  }
+
+  @Override
+  protected String[] getRow(int rowNum) {
+    if (rowNum < completed.getSummaryAsList().size()) {
+      Map<String, Map<String, BigDecimal>> completedSummary = completed.getSummaryAsList().get(rowNum);
+      return makeSummaryRow(completedSummary);
+    }
+    rowNum -= completed.getSummaryAsList().size();
+    
+    if (rowNum == 0) {
+      return makeBlankRow();
+    } else if (rowNum == 1) {
+      return makeHeadingRow(getDetailedHeadings());
+    }
+    rowNum -= 2;
+
+    if (rowNum < completed.getDetailedData().size()) {
+      return makeDetailedRow(completed.getDetailedData().get(rowNum));
+    }
+    rowNum -= completed.getDetailedData().size();
+    
+    if (rowNum == 0) {
+      return makeBlankRow();
+    } else if (rowNum == 1) {
+      return makeHeadingRow(getColumns().stream().map(th -> th.getHeading()).collect(Collectors.toList()));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)